Source/DafnyPipeline.Test/ProverLogStability.cs

```diff
@@ -0,0 +1,109 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using DafnyCore.Test;
+using DafnyTestGeneration;
+using Microsoft.Boogie;
+using Microsoft.Dafny;
+using Xunit;
+using Xunit.Abstractions;
+using BoogieProgram = Microsoft.Boogie.Program;
+
+namespace DafnyPipeline.Test {
+  // Main.Resolve has static shared state (TypeConstraint.ErrorsToBeReported for example)
+  // so we can't execute tests that use it in parallel.
+  [Collection("Singleton Test Collection - Resolution")]
+  public class ProverLogStabilityTest {
+
+    private readonly ITestOutputHelper testOutputHelper;
+
+    // All types of top level declarations.
+    readonly string originalProgram = @"
```

Comment:
This is a good start at least, but I feel like it's likely one of the integration test files would cover more features, especially some non-trivial assertion verification.